bolt/lib/Core/BinaryFunction.cpp

```diff
@@ -4366,6 +4367,10 @@ MCInst *BinaryFunction::getInstructionAtOffset(uint64_t Offset) {
   }
 }
 
+bool BinaryFunction::shouldPreserveNops() const {
+  return PreserveNops || opts::KeepNops;
```

Comment:
Small nit: Maybe add option check to RemoveNops::runOnFunctions too/instead?

Reply:
What's the rationale behind that extra check?

Reply:
Not to loop though the functions in pass. It is minor thing

Reply:
Ah - we can skip the pass all over: #72228.